### PR TITLE
[JENKINS-29347] Handle Git mergeStategy parameter

### PR DESCRIPTION
**Pull request by aduprat**
I don't want to break current mergeOptions signature, so i introduce two methods in order to get rid of the first optional parameter.

### CODE REVIEW

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/scm/GitContext.groovy [hunk 1]
```diff
@@ -66,18 +66,26 @@ class GitContext extends AbstractContext {
         executeInContext(strategyClosure, strategyContext)
     }
 
-    void mergeOptions(String remote = null, String branch) {
```
Comment (daspilker) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/scm/GitContext.groovy]: Leave the existing method as it is, there is no need to change it.

**Comment by daspilker** [comment]
And update the documentation.

File: job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/scm/GitContext.groovy (hunk 1 above)
Comment (aduprat) [inline job-dsl-core/src/main/groovy/javaposse/jobdsl/dsl/helpers/scm/GitContext.groovy]: OK

**Comment by aduprat** [comment]
Documentation updated.

**Comment by daspilker** [comment]
Thanks, I rebased your changes and merged them manually.

**Pull request closed without merging**